src/ResourceManager/Sql/Commands.Sql/ManagedDatabase/Cmdlet/RestoreAzureSqlManagedDatabase.cs

```diff
@@ -0,0 +1,174 @@
+// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Sql.ManagedDatabase.Model;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using System;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+namespace Microsoft.Azure.Commands.Sql.ManagedDatabase.Cmdlet
+{
+    [Cmdlet(VerbsData.Restore, "AzureRmSqlManagedDatabase",
+        SupportsShouldProcess = true),
+        OutputType(typeof(AzureSqlManagedDatabaseModel))]
+    public class RestoreAzureRmSqlManagedDatabase
+        : AzureSqlManagedDatabaseCmdletBase<AzureSqlManagedDatabaseModel>
+    {
+        protected const string PointInTimeRestoreFromNameAndResourceGroupParameterSet =
+            "PointInTimeRestoreManagedDatabaseFromInputParameters";
+
+        protected const string PointInTimeRestoreFropmInputObjectParameterSet =
+            "PointInTimeRestoreManagedDatabaseFromAzureSqlManagedDatabaseModelInstanceDefinition";
+
+        protected const string PointInTimeRestoreFromResourceIdParameterSet =
+            "PointInTimeRestoreManagedDatabaseFromAzureResourceId";
+
+        /// <summary>
+        /// Gets or sets flag indicating a restore from a point-in-time backup.
+        /// </summary>
+        [Parameter(
+            ParameterSetName = PointInTimeRestoreFromNameAndResourceGroupParameterSet,
+            Mandatory = true,
+            HelpMessage = "Restore from a point-in-time backup.")]
+        [Parameter(
+            ParameterSetName = PointInTimeRestoreFropmInputObjectParameterSet,
+            Mandatory = true,
+            HelpMessage = "Restore from a point-in-time backup.")]
+        [Parameter(
+            ParameterSetName = PointInTimeRestoreFromResourceIdParameterSet,
+            Mandatory = true,
+            HelpMessage = "Restore from a point-in-time backup.")]
+        public SwitchParameter FromPointInTimeBackup { get; set; }
```

Comment:
Put this below ResourceId so it doesn't show up first in the syntax.  Also, this shouldn't be mandatory, should it?

Reply:
Actually, this should be mandatory I guess. Currently there is only one type of restore, so this one is kind of redundant, but I am planing to add in future some other types of restore and switch parameter should be mandatory to distinguish which restore to do. Also having it in start looks better to me (I am using the simmilar approach as AzureSqlDbrestore cmdlet)